RespokeSDKTests/Functional/ConnectionDisconnectTests.m

```diff
@@ -0,0 +1,242 @@
+{
+    // Create a client to test with
+    NSString *recipientEndpointID = [RespokeTestCase generateTestEndpointID];
+    
+    // Create a second client to test with
+    NSString *senderEndpointID = [RespokeTestCase generateTestEndpointID];
+    RespokeClient *senderClient = [self createTestClientWithEndpointID:senderEndpointID delegate:self];
+    
+    // Join Group
+    asyncTaskDone = NO;
+    NSString *testGroupID = [@"group" stringByAppendingString:recipientEndpointID];
+    [senderClient joinGroups:@[testGroupID] successHandler:^(NSArray *groups){
+        firstClientGroup = [groups firstObject];
+        asyncTaskDone = YES;
+    } errorHandler:^(NSString *errorMessage){
+        asyncTaskDone = YES;
+    }];
+    
+    [self waitForCompletion:TEST_TIMEOUT];
+    
+    // diconnect and immediately reconnect
+    [senderClient disconnect];
+    [self createTestClientWithEndpointID:senderEndpointID delegate:self];
+    
+
+    // message should error out
+    asyncTaskDone = NO;
+    callbackDidSucceed = NO;
+    [firstClientGroup sendMessage:TEST_MESSAGE push:NO persist:NO successHandler:^{
+        callbackDidSucceed = NO;
+        asyncTaskDone = YES;
+        XCTAssertTrue(NO, @"Should not successfully send a group message.");
+    } errorHandler:^(NSString *errorMessage){
+        NSLog(@"Error Message: %@", errorMessage);
+        callbackDidSucceed = YES;
+        asyncTaskDone = YES;
+    }];
+    
+    [self waitForCompletion:TEST_TIMEOUT];
+    XCTAssertTrue(callbackDidSucceed, @"sendMessage should call successHandler");
+}
+
+/**
+ *  This test should confirm that a null object is returned when executing getGroupWithID to create a RespokeGroup from a group Id String
+ */
+- (void)testErrorWhenReconnectingGroupMessageGetGroupWithID
+{
+    // Create a client to test with
+    NSString *recipientEndpointID = [RespokeTestCase generateTestEndpointID];
+    
+    // Create a second client to test with
+    NSString *senderEndpointID = [RespokeTestCase generateTestEndpointID];
+    RespokeClient *senderClient = [self createTestClientWithEndpointID:senderEndpointID delegate:self];
+    
+    // Join Group
+    asyncTaskDone = NO;
+    NSString *testGroupID = [@"group" stringByAppendingString:recipientEndpointID];
+    [senderClient joinGroups:@[testGroupID] successHandler:^(NSArray *groups){
+        firstClientGroup = [groups firstObject];
+        asyncTaskDone = YES;
+    } errorHandler:^(NSString *errorMessage){
+        asyncTaskDone = YES;
+    }];
+    
+    [self waitForCompletion:TEST_TIMEOUT];
+
+    // diconnect and immediately reconnect
+    [senderClient disconnect];
+    [self createTestClientWithEndpointID:senderEndpointID delegate:self];
+    
+    RespokeGroup *respokeGroup = [senderClient getGroupWithID:testGroupID];
+    XCTAssertNil(respokeGroup, @"Should be a null value");
+}
+
+#pragma mark - RespokeClientDelegate methods
+
+
+- (void)onConnect:(RespokeClient*)sender
+{
+    asyncTaskDone = YES;
```

Comment:
I could be wrong but I don't think you need `asyncTaskDone = YES;` here since the createTestClientWithEndpointID handles blocking while the connect is happening.